File: js/encryption/symmetricEncryption.js
// symmetricEncryption.js

const crypto = require('crypto');

// Key and IV (Initialization Vector) for AES encryption
const algorithm = 'aes-256-cbc';
const key = crypto.randomBytes(32); // 256-bit key
const iv = crypto.randomBytes(16);  // 128-bit IV

// Encrypt function
function encrypt(text) {
  const cipher = crypto.createCipheriv(algorithm, key, iv);
  let encrypted = cipher.update(text, 'utf8', 'hex');
  encrypted += cipher.final('hex');
  return { encryptedData: encrypted, iv: iv.toString('hex') };
}

// Decrypt function
function decrypt(encryptedData, ivHex) {
  const iv = Buffer.from(ivHex, 'hex');
  const decipher = crypto.createDecipheriv(algorithm, key, iv);
  let decrypted = decipher.update(encryptedData, 'hex', 'utf8');
  decrypted += decipher.final('utf8');
  return decrypted;
}

module.exports = { encrypt, decrypt };

File: python/encryption/symmetric_encryption.py
# symmetric_encryption.py

from cryptography.fernet import Fernet

# Generate a key
key = Fernet.generate_key()
cipher = Fernet(key)

# Encrypt function
def encrypt(message: str) -> bytes:
    encrypted = cipher.encrypt(message.encode())
    return encrypted

# Decrypt function
def decrypt(encrypted_message: bytes) -> str:
    decrypted = cipher.decrypt(encrypted_message).decode()
    return decrypted

if __name__ == "__main__":
    msg = "Hello, World!"
    encrypted_msg = encrypt(msg)
    print(f"Encrypted: {encrypted_msg}")
    print(f"Decrypted: {decrypt(encrypted_msg)}")
